store/inmem/InMem.go

```diff
@@ -56,17 +70,23 @@ func (i *InMem) Get(key model.Key) (store.OwnableItem, error) {
 	if !ok {
 		return store.OwnableItem{}, store.SanitizeError(store.ItemOperationError{Err: store.ErrItemNotFound, Key: key, Operation: "get"})
 	}
-	return item, nil
+	if i.itemExpired(item) {
+		i.pruneExpiredItem(key.Bucket, key.ID, bucket)
+		return store.OwnableItem{}, store.SanitizeError(store.ItemOperationError{Err: store.ErrItemNotFound, Key: key, Operation: "get"})
+	}
+
+	return item.OwnableItem, nil
 }
 
 func (i *InMem) GetAll(bucket string) (map[string]store.OwnableItem, error) {
 	i.lock.RLock()
 	defer i.lock.RUnlock()
 	items := i.data[bucket]
 	if items == nil {
-		items = map[string]store.OwnableItem{}
+		items = map[string]expireableItem{}
 	}
-	return items, nil
+	i.pruneExpiredItems(bucket, items)
```

Comment:
Since we only have a read lock, I don't think we can prune here?

Reply:
Good catch! Not sure how the tests here were passing when the first time I ran `go test -race` failed immediately locally. I added a fix and also some documentation explicitly stating this is not recommended for production environments.